File: Backend/Models/Showtime.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class Showtime
    {
        public int Id { get; set; }

        [Required]
        [ForeignKey("Movie")]
        public int MovieId { get; set; }
        public Movie? Movie { get; set; } // Liên kết với phim

        [Required]
        [ForeignKey("Theatre")]
        public int TheatreId { get; set; }
        public Theatre? Theatre { get; set; } // Liên kết với rạp

        [Required]
        public string Screen { get; set; } = string.Empty; // Màn hình chiếu

        [Required]
        public DateTime StartTime { get; set; } // Thời gian bắt đầu

        [NotMapped]
        public DateTime EndTime => Movie != null ? StartTime.AddMinutes(Movie.DurationInMinutes) : StartTime;

        [Required]
        public string Format { get; set; } = "2D"; // Định dạng suất chiếu (2D, 3D, IMAX)

        [Required]
        public string ShowType { get; set; } = "Phụ đề"; // Loại suất chiếu (Phụ đề, Lồng tiếng)

        [Required]
        public int SeatCapacity { get; set; } = 200; // Tổng số ghế trong suất chiếu

        public decimal TicketPrice { get; set; } = 100000; // Giá vé (VNĐ)

        [Required]
        public string Status { get; set; } = "Active"; // Trạng thái suất chiếu (Active, Cancelled)
    }
}

File: Frontend/lib/models/showtime.dart
import 'package:bookingmovieticket/models/movie_model.dart';

class Showtime {
  final int id;
  final int movieId;
  final int theatreId;
  final String screen;
  final DateTime startTime;
  final DateTime endTime;
  final String format;
  final String showType;
  final int seatCapacity;
  final double ticketPrice;
  final String status;
  final int screenId;
  final Movie? movie;

  Showtime({
    required this.id,
    required this.movieId,
    required this.theatreId,
    required this.screen,
    required this.startTime,
    required this.endTime,
    required this.format,
    required this.showType,
    required this.seatCapacity,
    required this.ticketPrice,
    required this.status,
    required this.screenId,
    this.movie,
  });

  factory Showtime.fromJson(Map<String, dynamic> json) {
    return Showtime(
      id: json['Id'] ?? 0,
      movieId: json['MovieId'] ?? 0,
      theatreId: json['TheatreId'] ?? 0,
      screen: json['Screen'] ?? 'N/A',
      startTime: DateTime.tryParse(json['StartTime'] ?? '') ?? DateTime.now(),
      endTime: DateTime.tryParse(json['EndTime'] ?? '') ?? DateTime.now(),
      format: json['Format'] ?? 'Unknown',
      showType: json['ShowType'] ?? 'Unknown',
      seatCapacity: json['SeatCapacity'] ?? 0,
      ticketPrice: (json['TicketPrice'] as num?)?.toDouble() ?? 0.0,
      status: json['Status'] ?? 'Inactive',
      screenId: json['ScreenId'] ?? 0,
      movie: json['Movie'] != null ? Movie.fromJson(json['Movie']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'Id': id,
      'MovieId': movieId,
      'TheatreId': theatreId,
      'Screen': screen,
      'StartTime': startTime.toIso8601String(),
      'EndTime': endTime.toIso8601String(),
      'Format': format,
      'ShowType': showType,
      'SeatCapacity': seatCapacity,
      'TicketPrice': ticketPrice,
      'Status': status,
      'ScreenId': screenId,
      'Movie': movie?.toJson(),
    };
  }
}
